doc/api/deprecations.md

```diff
@@ -3019,6 +3019,20 @@ should have the same effect. The receiving end should also check the
 [`readable.readableEnded`][] value on [`http.IncomingMessage`][] to get whether
 it was an aborted or graceful destroy.
 
+### DEP0157: Thenable support in streams
+
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/40860
+    description: Documentation-only deprecation.
+-->
+
+Type: Documentation-only
+
+An undocumented feature of Node.js streams was to support thenables in
+implementation methods. This is now deprecated, instead use callbacks.
```

Comment:
It would be better to mention _why_ we deprecated this.

Reply:
@ranag can you help please? Would the following work?
```suggestion
implementation methods. This feature was causing stream instability when using
async functions as stream methods. This is now deprecated, instead use
callbacks.
```

Reply:
The text work for me

Reply:
I would also suggest adding an example of what is being deprecated as it's not very clear